workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/SpatialOSBehaviourLibrary.cs

```diff
@@ -0,0 +1,191 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core
+{
+    /// <summary>
+    ///     Retrieves Reader and Writer fields from MonoBehaviours and handles injection into them.
+    /// </summary>
+    public class SpatialOSBehaviourLibrary
+    {
+        private readonly Dictionary<Type, Dictionary<uint, FieldInfo>> fieldHandleCache
+            = new Dictionary<Type, Dictionary<uint, FieldInfo>>();
+
+        private readonly Dictionary<Type, List<uint>> componentReaderIdsForBehaviours =
+            new Dictionary<Type, List<uint>>();
+
+        private readonly Dictionary<Type, List<uint>> componentWriterIdsForBehaviours =
+            new Dictionary<Type, List<uint>>();
+
+        private readonly HashSet<Type> invalidMonoBehaviourTypes = new HashSet<Type>();
+
+        private readonly ILogDispatcher logger;
+        private const string LoggerName = "SpatialOSBehaviourLibrary";
+
+        private const string BadRequiredMemberWarning
+            = "[Require] attribute found on member that is not Reader or Writer or incorrectly generated, ignoring this member!";
```

Comment:
It might make sense to change the wording to something more formal, e.g. change `... generated, ignoring this member!` to `... generated. This member will be ignored.`. Also, leave out the `incorrectly generated` part. Just say that it's neither a reader nor writer.

Reply:
Also, how about changing the name "Bad..." to something else, like `InvalidRequiredMemberWarning`?